flink-tests/src/test/java/org/apache/flink/test/classloading/jar/CheckpointedStreamingProgram.java

```diff
@@ -72,34 +74,27 @@ public void run(SourceContext<String> ctx) throws Exception {
 		public void cancel() {
 			running = false;
 		}
-
-		@Override
-		public Integer snapshotState(long checkpointId, long checkpointTimestamp) throws Exception {
-			return null;
-		}
-
-		@Override
-		public void restoreState(Integer state) {
-
-		}
 	}
 
-	public static class StatefulMapper implements MapFunction<String, String>, Checkpointed<StatefulMapper>, CheckpointListener {
+	public static class StatefulMapper implements MapFunction<String, String>, ListCheckpointed<StatefulMapper>, CheckpointListener {
 
 		private String someState;
 		private boolean atLeastOneSnapshotComplete = false;
 		private boolean restored = false;
 
 		@Override
-		public StatefulMapper snapshotState(long checkpointId, long checkpointTimestamp) throws Exception {
-			return this;
+		public List<StatefulMapper> snapshotState(long checkpointId, long timestamp) throws Exception {
+			return Collections.singletonList(this);
 		}
 
 		@Override
-		public void restoreState(StatefulMapper state) {
-			restored = true;
-			this.someState = state.someState;
-			this.atLeastOneSnapshotComplete = state.atLeastOneSnapshotComplete;
+		public void restoreState(List<StatefulMapper> state) throws Exception {
+			if (!state.isEmpty()) {
```

Comment:
If the state is empty we should fail immediately; currently (I think) this would cause us to fail with the RuntimeException saying "Intended failure, to trigger restore", which is a bit inaccurate.